src/common/envalid.ts

```diff
@@ -18,19 +18,36 @@ const cleanSpec = {
   VALUES_INPUT: str({ desc: 'The chart values.yaml file', default: undefined }),
 }
 
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-export const cleanEnvironment = () => {
-  let pEnv: any = process.env
-  const path = `${pEnv.ENV_DIR}/.secrets`
-  if (pEnv.ENV_DIR && existsSync(path)) {
-    const result = config({ path }) // this sets vars from .env onto process.env
-    if (result.error) console.error(result.error)
-    pEnv = { ...pEnv, ...result.parsed }
+export const taskEnvSpec = {
+  OTOMI_FLAGS: json({ default: {} }),
+  TEAM_IDS: json({ desc: 'A list of team ids in JSON format' }),
+}
+
+export const cleanEnvironment = (
+  spec: Record<string, any> = cliEnvSpec,
+  returnFunc = false,
+): Record<string, any> | CallableFunction => {
+  const func = (): Record<string, any> => {
+    let pEnv: any = process.env
+    // load local .env if we have it, for devs
+    let path = `${process.cwd()}/.env`
+    if (existsSync(path)) {
+      const result = config({ path })
+      if (result.error) console.error(result.error)
+      pEnv = { ...pEnv, ...result.parsed }
+    }
+    path = `${pEnv.ENV_DIR}/.secrets`
+    if (existsSync(path)) {
+      const result = config({ path })
+      if (result.error) console.error(result.error)
+      pEnv = { ...pEnv, ...result.parsed }
+    }
+    return cleanEnv(pEnv, spec)
   }
-  return cleanEnv(pEnv, cleanSpec)
+  return returnFunc ? func : func()
 }
 
-export const env = cleanEnvironment()
+export const env = cleanEnvironment() as Record<string, any>
 
 export const isChart = env.CI && !!env.VALUES_INPUT
```

Comment:
I am confused why these bools are not part of env, now we have to skip the TS check using `as` in L50.